plugins/outputs/signalfx/signalfx.go

```diff
@@ -0,0 +1,302 @@
+package signalfx
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/golib/v3/datapoint/dpsink"
+	"github.com/signalfx/golib/v3/event"
+	"github.com/signalfx/golib/v3/sfxclient"
+	"github.com/signalfx/signalfx-go/realm"
+	"github.com/signalfx/signalfx-go/writer"
+)
+
+/*init initializes the plugin context*/
+func init() {
+	outputs.Add("signalfx", func() telegraf.Output {
+		return NewSignalFx()
+	})
+}
+
+/*SignalFx plugin context*/
+type SignalFx struct {
+	AccessToken        string   `toml:"access_token"`
+	BatchSize          int      `toml:"batch_size"`
+	MaxBuffered        int      `toml:"max_buffered"`
+	SignalFxRealm      string   `toml:"signalfx_realm"`
+	IngestURL          string   `toml:"ingest_url"`
+	IncludedEventNames []string `toml:"included_event_names"`
+
+	includedEventSet map[string]bool
+	client           dpsink.Sink
+	dps              chan []*datapoint.Datapoint
+	evts             chan *event.Event
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+var sampleConfig = `
+    ## SignalFx Org Access Token
+    access_token = "my-secret-token"
+
+    ## Maximum batch size of datapoints to send to ingest at once
+    batch_size = 1000
+
+	## Maximum number of datapoints or events to buffer in the agent before dropping data
+	max_buffered = 100000
```

Comment:
The library used by this PR doesn't integrate very well with Telegraf, as it wants to take over responsibilities that Telegraf typically controls.

For example, batch size and max bufferer are essentially the same as theTelegraf `metric_batch_size` and `metric_buffer_limit` options.   The Write method should be able to indicate when a write has failed/succeeded so that Telegraf can manage retries and know when collected data has been written.

I think we should consider removing the SignalFX library and writing a simple client using the `http` package instead.

Reply:
Ok, I simplified the plugin to synchronously send using our low-level client library.  Removed the now unneeded config in the plugin.

Reply:
@danielnelson is the way I did it good?